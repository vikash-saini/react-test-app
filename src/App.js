import logo from "./logo.svg";
import "./App.css";
import { useState } from "react";

function App() {
  const [name, setName] = useState("");
  const [data, setData] = useState("");
  return (
    <div className="App">
      <img
        height="200px"
        src="https://cdn.siasat.com/wp-content/uploads/2023/05/RK-AI-780x470.jpg"
        alt="AI generated Image"
      />
      <br />
      <p>let's code</p>
      <br />
      <input
        type="text"
        placeholder="enter username"
        name="Username"
        id="userId"
        onChange={(e) => setName(e.target.value)}
      />
      <br />
      Click me{" "}
      <button onClick={() => setData("updated data")}>update Code</button>
      <br />
      <h2>{data}</h2>
    </div>
  );
}

export default App;
