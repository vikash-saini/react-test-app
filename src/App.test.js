import { fireEvent, render, screen } from "@testing-library/react";
import App from "./App";

// beforeAll(() => {
//   console.log("***This will trigger before all tests***");
// });

// beforeEach(() => {
//   console.log("***This will trigger before each test***");
// });

// test("renders learn react link", () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });
test("app componenets screen test", () => {
  render(<App />);
  const ele = screen.getByText("let's code");
  const altText = screen.getByAltText("AI generated Image");
  expect(ele).toBeInTheDocument();
  expect(altText).toBeInTheDocument();
});

test("testing input box", () => {
  render(<App />);
  let checkInput = screen.getByRole("textbox");
  let checkPlaceholder = screen.getByPlaceholderText(/enter username/i);
  expect(checkPlaceholder).toBeInTheDocument();
  expect(checkInput).toBeInTheDocument();
  expect(checkInput).toHaveAttribute("name", "Username");
  expect(checkInput).toHaveAttribute("id", "userId");
});

test("general testing", () => {
  render(<App />);
  const checkInput = screen.getByRole("textbox");
  expect(checkInput).toBeInTheDocument();
  expect(checkInput).toHaveAttribute("placeholder", "enter username");
});

describe.skip("testiung  UI", () => {
  test("testing input box", () => {
    render(<App />);
    let checkInput = screen.getByRole("textbox");
    let checkPlaceholder = screen.getByPlaceholderText(/enter username/i);
    expect(checkPlaceholder).toBeInTheDocument();
    expect(checkInput).toBeInTheDocument();
    expect(checkInput).toHaveAttribute("name", "Username");
    expect(checkInput).toHaveAttribute("id", "userId");
  });

  test("general testing", () => {
    render(<App />);
    const checkInput = screen.getByRole("textbox");
    expect(checkInput).toBeInTheDocument();
    expect(checkInput).toHaveAttribute("placeholder", "enter username");
  });
});

test("input Event testing", () => {
  render(<App />);
  const input = screen.getByRole("textbox");
  fireEvent.change(input, { target: { value: "abc" } });
  expect(input.value).toBe("abc");
});

test("button click test", () => {
  render(<App />);
  let btn = screen.getByRole("button");
  fireEvent.click(btn);
  expect(screen.getByText(/updated data/i)).toBeInTheDocument();
});

afterAll(() => {
  console.log("***This will trigger after all tests***");
});

// afterEach(()=>{
//   console.log("***This will trigger before each test***");
// })

test("snapshot testing",()=>{
  const container = render(<App/>);
  expect(container).toMatchSnapshot();
})