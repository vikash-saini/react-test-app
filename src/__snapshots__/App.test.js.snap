// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <img
          alt="AI generated Image"
          height="200px"
          src="https://cdn.siasat.com/wp-content/uploads/2023/05/RK-AI-780x470.jpg"
        />
        <br />
        <p>
          let's code
        </p>
        <br />
        <input
          id="userId"
          name="Username"
          placeholder="enter username"
          type="text"
        />
        <br />
        Click me
         
        <button>
          update Code
        </button>
        <br />
        <h2 />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <img
        alt="AI generated Image"
        height="200px"
        src="https://cdn.siasat.com/wp-content/uploads/2023/05/RK-AI-780x470.jpg"
      />
      <br />
      <p>
        let's code
      </p>
      <br />
      <input
        id="userId"
        name="Username"
        placeholder="enter username"
        type="text"
      />
      <br />
      Click me
       
      <button>
        update Code
      </button>
      <br />
      <h2 />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
